### PR TITLE
NotificationsManager: Revamp Mark I

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Description:
This is a multipart revamp. In this PR we're splitting part of `NotificationsManager` functionality into `PushNotificationsManager` and `InteractiveNotificationsHandler`. The remaining methods will be migrated over in other branches.

**Important:** please, review / merge #4557 first, since this branch includes those changes.
Reference: #4069

#### Scenario 1: Registration
1. Fresh install WPiOS
2. Log into your account and enable Push Notifications

Verify that the PN's are effectively received.

#### Scenario 2: Un-registration
1. Right after testing Scenario 1, logout from your account

Verify that PN's are not received anymore.

#### Scenario 3: Disabled Notifications
1. Open Settings.app
2. Open WordPress Settings
3. Disable Push Notifications
4. Launch WPiOS
5. Open `Me` > (Any blog) > `Push Notifications`

Verify that the app detects, effectively, that APN's are disabled.

#### Scenario 4: Comment Actions
1. Kill WPiOS
2. Receive a comment-Y notification
3. Pull over iOS Notification to reveal Interactive Actions.

Verify that you get `Approve` (OR) `Reply / Like`, depending on your blog's discussion settings.

--

Needs Review: @astralbodies 
Aaron, since you're also taking over #4557's review, may i also bother you with this one?. I'd be happy to fire the Development Environment APN's server, please, just let me know.

Thank you!

### CODE REVIEW

**Comment by astralbodies** [comment]
Reviewing now @jleandroperez

**Comment by astralbodies** [comment]
Functionality passed testing - going to review code now.

File: WordPress/Classes/Utility/InteractiveNotificationsHandler.swift [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import Foundation
+
+
+/**
+ *  @extension      InteractiveNotificationsHandler
+ *  @details        In this class, we'll encapsulate all of the code related to UIUserNotificationCategory and 
+ *                  UIUserNotificationAction instantiation, along with the required handlers.
+ */
+
+final public class InteractiveNotificationsHandler
```
Comment (astralbodies) [inline WordPress/Classes/Utility/InteractiveNotificationsHandler.swift]: I'm digging the final-first approach on classes

File: WordPress/Classes/Utility/PushNotificationsManager.swift [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                DDLogSwift.logInfo("Successfully unregistered Device ID \(knownDeviceId) for Push Notifications!")
+                
+                self.deviceToken = nil
+                self.deviceId = nil
+                
+                let notificationCenter = NSNotificationCenter.defaultCenter()
+                notificationCenter.postNotificationName(NotificationsManagerDidUnregisterDeviceToken, object: nil)
+                
+            },
+            failure: { (error: NSError) -> Void in
```
Comment (astralbodies) [inline WordPress/Classes/Utility/PushNotificationsManager.swift]: technically you can leave the Void thing out:
```swift
failure: { error: NSError in
```

**Comment by astralbodies** [comment]
This all looks really great @jleandroperez! There are bits that sort of went over my head but I trust your design. It feels very solid!
:shipit:

File: WordPress/Classes/Utility/PushNotificationsManager.swift (hunk 2 above)
Comment (jleandroperez) [inline WordPress/Classes/Utility/PushNotificationsManager.swift]: *Love* that touch, i had no idea that was valid. Thank you!!

**Comment by jleandroperez** [comment]
Thanks **a lot** for the review Aaron!

**Pull request merged**